File: srclab-common/src/main/java/xyz/srclab/common/base/ToString.java
package xyz.srclab.common.base;

import xyz.srclab.annotation.Nullable;

import java.util.Arrays;

/**
 * @author sunqian
 */
public class ToString {

    public static String toString(@Nullable Object any) {
        if (any == null) {
            return String.valueOf((Object) null);
        }
        return deepToString(any);
    }

    private static String deepToString(Object any) {
        if (any instanceof Object[]) {
            return Arrays.deepToString((Object[]) any);
        }
        if (any instanceof boolean[]) {
            return Arrays.toString((boolean[]) any);
        }
        if (any instanceof byte[]) {
            return Arrays.toString((byte[]) any);
        }
        if (any instanceof short[]) {
            return Arrays.toString((short[]) any);
        }
        if (any instanceof char[]) {
            return Arrays.toString((char[]) any);
        }
        if (any instanceof int[]) {
            return Arrays.toString((int[]) any);
        }
        if (any instanceof long[]) {
            return Arrays.toString((long[]) any);
        }
        if (any instanceof float[]) {
            return Arrays.toString((float[]) any);
        }
        if (any instanceof double[]) {
            return Arrays.toString((double[]) any);
        }
        return any.toString();
    }
}

File: srclab-common/src/main/kotlin/xyz/srclab/common/base/ToString.kt
package xyz.srclab.common.base

import java.util.*

/**
 * @author sunqian
 */
object ToString {

    @JvmStatic
    fun toString(any: Any?): String {
        return any.toString()
    }

    @JvmStatic
    fun deepToString(any: Any?): String {
        if (any == null) {
            return "null"
        }
        if (any is Array<*>) {
            return Arrays.deepToString(any)
        }
        if (any is BooleanArray) {
            return Arrays.toString(any)
        }
        if (any is ByteArray) {
            return Arrays.toString(any)
        }
        if (any is ShortArray) {
            return Arrays.toString(any)
        }
        if (any is CharArray) {
            return Arrays.toString(any)
        }
        if (any is IntArray) {
            return Arrays.toString(any)
        }
        if (any is LongArray) {
            return Arrays.toString(any)
        }
        if (any is FloatArray) {
            return Arrays.toString(any)
        }
        if (any is DoubleArray) {
            return Arrays.toString(any)
        }
        return any.toString()
    }
}